src/child/actions/favourites.js

```diff
@@ -17,3 +17,24 @@ export const quandlResponse = createActionCreator((code, name) => ({
     code,
     name
 }));
+
+const removeFavourite = createActionCreator((dragStartWindow, code) => ({
+    type: ACTION_TYPES.REMOVE_FAVOURITE,
+    dragStartWindow,
+    code
+}));
+
+export function moveFavouriteFromWindow(dragStartWindow, code) {
+    return (dispatch, getState) => {
+        if (getState().childWindows[dragStartWindow].favourites.codes.length > 1) {
+            dispatch(removeFavourite(dragStartWindow, code));
```

Comment:
Maybe I'm oversimplifying this, but wouldn't simply toggling the favourite in the `dragStartWindow` window solve this and remove all of the logic in the childWindows reducer below?

We'd need to add a new action creator to specify the window to toggle the favourite in, but that should be pretty simple:

```js
function toggleFavouriteInWindow(windowName, code) {
    return {
        type: ACTION_TYPES.TOGGLE_FAVOURITE,
        windowName,
        code
    };
}
```

Then `dispatch(toggleFavouriteInWindow(dragStartWindow, code))` instead of `dispatch(removeFavourite(dragStartWindow, code))`.

The else should handle closing the window and we already have the `CLOSE` action in place to handle this and delete the corresponding window state.